Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -60,33 +60,34 @@ extension HTTPClient {
     }
 
     public struct Request {
-        public var version: HTTPVersion
-        public var method: HTTPMethod
-        public var url: URL
-        public var scheme: String
-        public var host: String
+        public let version: HTTPVersion
+        public let method: HTTPMethod
+        public let url: URL
         public var headers: HTTPHeaders
         public var body: Body?
+        
+        internal let scheme: String
+        internal let host: String
 
-        public init(url: String, version: HTTPVersion = HTTPVersion(major: 1, minor: 1), method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: Body? = nil) throws {
+        public init?(url: String, version: HTTPVersion = HTTPVersion(major: 1, minor: 1), method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: Body? = nil) {
             guard let url = URL(string: url) else {
-                throw HTTPClientError.invalidURL
+                return nil
             }
 
-            try self.init(url: url, version: version, method: method, headers: headers, body: body)
+            self.init(url: url, version: version, method: method, headers: headers, body: body)
         }
 
-        public init(url: URL, version: HTTPVersion = HTTPVersion(major: 1, minor: 1), method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: Body? = nil) throws {
-            guard let scheme = url.scheme else {
-                throw HTTPClientError.emptyScheme
+        public init?(url: URL, version: HTTPVersion = HTTPVersion(major: 1, minor: 1), method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: Body? = nil) {
+            guard let scheme = url.scheme?.lowercased() else {
```

Comment:
I do agree that this is much nicer as a user, but we're sacrificing the type of error for a "this worked or not" without the context on what the problem was. I think I'd prefer to keep the granular errors vs. checking for `nil` instead, but if there's already been discussion (or others have the other preference!) happy to defer.

Reply:
I didn't follow all of the discussions here recently... but would it be an option to add address the above concern by adding `static func validated(<mirror init's param list>) throws -> Request`, users get to choose then, and if it was discussed that the `nil` is "nicer" then that's indeed the default still 🤔

Reply:
Great call, that helps with surfacing the specific issue if someone really wants it. 🧠

Reply:
I find `nil`-returning functions like this *profoundly* frustrating as they're almost impossible to debug. It's very annoying to look at something you're confident is valid data and to have the program tell you that it is not valid, without any explanation as to why.

I'd be strongly -1 on this without at least the option for a constructor that gives me better diagnostics, but frankly I'm not sure if the convenience of the `nil` return is worth it for the inevitable failure of debuggability down the line.

Reply:
@Lukasa this PR originated as a simpler and safer way compared to #43. totally agree on the information loss point, and think we can keep the API throwing. that said, the exiting throwing code has a validation/mutability race that needs to be addressed and this PR demonstrates that part

Reply:
this is the throwing version: https://github.com/swift-server/async-http-client/pull/67